prime-dotnet-webapi/Controllers/GisController.cs

```diff
@@ -0,0 +1,210 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+using Prime.Services;
+using Prime.Models.Api;
+using Prime.ViewModels.Parties;
+
+namespace Prime.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class GisController : ControllerBase
+    {
+        private readonly IGisService _gisService;
+        public GisController(IGisService gisService)
+        {
+            _gisService = gisService;
+        }
+
+        // POST: api/gis
+        /// <summary>
+        /// Creates a new Gis Enrolment
+        /// </summary>
+        [HttpPost(Name = nameof(CreateGisParty))]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> CreateGisParty(GisChangeModel changeModel)
+        {
+            if (changeModel == null)
+            {
+                ModelState.AddModelError("Party", "Could not create the Party, the passed in model cannot be null.");
+                return BadRequest(ApiResponse.BadRequest(ModelState));
+            }
+
+            if (!changeModel.Validate(User))
+            {
+                ModelState.AddModelError("GisEnrolment", "One or more Properties did not match the information on the card.");
+                return BadRequest(ApiResponse.BadRequest(ModelState));
+            }
+
+            var createdGisId = await _gisService.CreateOrUpdateGisEnrolmentAsync(changeModel, User);
+
+            var gisEnrolment = await _gisService.GetGisEnrolmentByIdAsync(createdGisId);
+
+            return CreatedAtAction(
+                nameof(GetGisEnrolmentById),
+                new { gisId = createdGisId },
+                ApiResponse.Result(gisEnrolment)
+            );
+        }
+
+        // PUT: api/Gis/5
+        /// <summary>
+        /// Updates a specific Gis Enrolment.
+        /// </summary>
+        /// <param name="gisId"></param>
+        /// <param name="changeModel"></param>
+        [HttpPut("{gisId}", Name = nameof(UpdateGisEnrollee))]
+        // [Authorize(Roles = Roles.PrimeEnrollee)]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<IActionResult> UpdateGisEnrollee(int gisId, GisChangeModel changeModel)
```

Comment:
These first three controller methods have different return types:
```C#
Task<ActionResult>
Task<IActionResult>
Task<ActionResult<GisViewModel>>
```
I feel we should change them all to Task<ActionResult> for consistency. We are not actually using the feature offered by typing our ActionResult(s) anywhere in our app.